commons/zenoh-codec/src/core/zint.rs

```diff
@@ -134,18 +144,20 @@ where
 
     fn read(self, reader: &mut R) -> Result<u64, Self::Error> {
         let mut b = reader.read_u8()?;
+        if (b & 0x80) == 0 {
+            return Ok(b as u64);
+        }
 
         let mut v = 0;
         let mut i = 0;
-        let mut k = VLE_LEN;
-        while b > 0x7f && k > 0 {
+        // 7 * VLE_LEN is beyond the maximum number of shift bits
+        while (b & 0x80) != 0 && i != 7 * VLE_LEN {
```

Comment:
should VLE_LEN be set to 9, and shall we keep 8 bits of data in the 8-th byte of buffer ? This would allow us to use 9 bytes at most for 8 + 7 * 8 = 64 bits, since we really don't care about continuation bit in the msB: we know it should be the last. Then the check  i != 7 * VLE_LEN can be replaced with I < 7* (VLE_LEN - 1) (i.e. I < 56).  As a bonus this would also allow us to never overflow u64, because we are using exactly 64 bits rather than 70.

Reply:
An update has been just committed to properly handle this case.